src/evm/P2pSigner.ts

```diff
@@ -135,15 +136,23 @@ class P2pSigner<TFactories extends RpcServiceFactoryMap = {}>
         return this.isLeader;
     }
 
-    public connectToChannel(channelId: Bytes) {
+    public async connectToChannel(channelId: Bytes) {
         this.setChannelId(channelId);
+
+        // Update status to NOT_OPENED/OPENED as soon as we know the channelId.
+        await this.p2pManager.stateManager.refreshOpenedStatusFromChain();
+
         return this.p2pManager.tryOpenConnectionToChannel(channelId.toString());
     }
 
     public disconnectFromPeers() {
         this.p2pManager.disconnectAll();
     }
 
+    public async getChannelStatus(): Promise<Status> {
+        return await this.p2pManager.stateManager.getChannelStatus();
+    }
+
```

Comment:
Same same, but different ;)
```suggestion
    public getChannelStatus(): Promise<Status> {
        return this.p2pManager.stateManager.getChannelStatus();
    }

```